skorch/tests/callbacks/test_early_stopping.py

```diff
@@ -0,0 +1,170 @@
+            self, net_clf_cls, broken_classifier_module, classifier_data,
+            early_stopping_cls):
+        patience = 5
+        max_epochs = 8
+        early_stopping_cb = early_stopping_cls(patience=patience)
+
+        net = net_clf_cls(
+            broken_classifier_module,
+            callbacks=[
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        bad_run = (
+            len(net.history) != max_epochs
+        )
+
+        assert bad_run
+
+    def test_custom_scoring_nonstop(
+        self, net_clf_cls, classifier_module, classifier_data,
+            early_stopping_cls, epoch_scoring_cls):
+        lower_is_better = False
+        scoring_name = 'valid_roc_auc'
+        patience = 5
+        max_epochs = 8
+        scoring_cb = epoch_scoring_cls(
+            'roc_auc', lower_is_better, name=scoring_name)
+        early_stopping_cb = early_stopping_cls(
+            patience=patience, lower_is_better=lower_is_better,
+            monitor=scoring_name)
+
+        net = net_clf_cls(
+            classifier_module,
+            callbacks=[
+                scoring_cb,
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        ok_run = (
+                len(net.history) == max_epochs
+        )
+        assert ok_run
+
+    def test_custom_scoring_stop(
+        self, net_clf_cls, broken_classifier_module, classifier_data,
+            early_stopping_cls, epoch_scoring_cls):
+        lower_is_better = False
+        scoring_name = 'valid_roc_auc'
+        patience = 5
+        max_epochs = 8
+        scoring_cb = epoch_scoring_cls(
+            'roc_auc', lower_is_better, name=scoring_name)
+        early_stopping_cb = early_stopping_cls(
+            patience=patience, lower_is_better=lower_is_better,
+            monitor=scoring_name)
+
+        net = net_clf_cls(
+            broken_classifier_module,
+            callbacks=[
+                scoring_cb,
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        stopped_run = (
+                len(net.history) != max_epochs
+        )
+        assert stopped_run
+
+    def test_stopping_big_absolute_threshold(
+            self, net_clf_cls, classifier_module, classifier_data,
+            early_stopping_cls):
+        patience = 5
+        max_epochs = 8
+        early_stopping_cb = early_stopping_cls(patience=patience,
+                                               threshold_mode='abs',
+                                               threshold=0.1)
+
+        net = net_clf_cls(
+            classifier_module,
+            callbacks=[
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        stopped_run = (
```

Comment:
Again, one-liner is enough here.

Also, as above, the should know that `assert len(net.history) == patience + 1 < max_epochs`.